platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/ToolingApiPolymorphismIntegrationTest.groovy

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache.isolated
+
+import org.gradle.api.Project
+import org.gradle.configurationcache.fixtures.BaseModel
+import org.gradle.configurationcache.fixtures.CompositeModel
+import org.gradle.configurationcache.fixtures.DeepChildModel
+import org.gradle.configurationcache.fixtures.ShallowChildModel
+import org.gradle.tooling.provider.model.ToolingModelBuilder
+
+class ToolingApiPolymorphismIntegrationTest extends AbstractIsolatedProjectsToolingApiIntegrationTest {
```

Comment:
:?: Why is this test located in `core-configuration/configuration-cache`?

Reply:
Aware, cleanup on Monday.